vault/activity_log.go

```diff
@@ -1641,9 +1654,20 @@ func (a *ActivityLog) precomputedQueryWorker() error {
 		}
 	}
 
-	// Delete the intent log
+	// delete the intent log
 	a.view.Delete(ctx, activityIntentLogKey)
 
+	// emit active entity metrics for the interval (times[0], endTime)
+	for nsID, counts := range byNamespace {
+		a.metrics.SetGaugeWithLabels(
+			[]string{"identity", "entity", "active", "monthly"},
```

Comment:
The comment is correct but I don't believe that is what this code does

Because we've finished the loop, byNamespace contains all the data from the interval (times[len(times)-1], endTime).  times is in reverse order with 0 being the most recent month.

A check at the same place you're doing the defaultReportingPeriod check would be sufficient.